apps/boxclk/app.js

```diff
@@ -207,195 +212,219 @@
-              boxesConfig[boxKey].boxPos.y = (boxPos[boxKey].y / h).toFixed(3);
-            });
-            storage.write(fileName, JSON.stringify(boxesConfig));
-            displaySaveIcon();
-            return;
-          }
-        } else {
-          // if any box is being dragged, just deselect all without saving
-          deselectAllBoxes();
-        }
-      }
-      if (Object.values(wasDragging).some(Boolean) || !boxTouched) {
-        draw(boxes);
-      }
-      doubleTapTimer = setTimeout(() => {
-        doubleTapTimer = null;
-      }, 500); // Increase or decrease this value based on the desired double tap timing
-      movementDistance = 0;
-    };
-
-    // ------------------------------------
-    // Define the dragHandler function
-    // ------------------------------------
-    dragHandler = function(e) {
-      // Check if any box is being dragged
-      if (!Object.values(isDragging).some(Boolean)) return;
-      // Calculate the movement distance
-      movementDistance += Math.abs(e.dx) + Math.abs(e.dy);
-      // Check if the movement distance exceeds a threshold
-      if (movementDistance > 1) {
-        boxKeys.forEach((boxKey) => {
-          if (isDragging[boxKey]) {
-            widgets.hide();
-            let boxItem = boxes[boxKey];
-            calcBoxSize(boxItem);
-            let newX = boxPos[boxKey].x + e.dx;
-            let newY = boxPos[boxKey].y + e.dy;
-            if (newX - totalWidth / 2 >= 0 &&
-                newX + totalWidth / 2 <= w &&
-                newY - totalHeight / 2 >= 0 &&
-                newY + totalHeight / 2 <= h ) {
-              boxPos[boxKey].x = newX;
-              boxPos[boxKey].y = newY;
-            }
-            const pos = calcBoxPos(boxKey);
-            g.clearRect(pos.x1, pos.y1, pos.x2, pos.y2);
-          }
-        });
-        draw(boxes);
+    Bangle.on('lock', function(isLocked) {
+      if (isLocked) {
+        // Screen is about to lock, deselect all boxes
+        deselectAllBoxes();
+        // Redraw to reflect changes
+        draw();
       }
-    };
+    });
 
     Bangle.on('touch', touchHandler);
     Bangle.on('drag', dragHandler);
 
+    if (boxes.step) {
+      boxes.step.string = formatStr(boxes.step, Bangle.getHealthStatus("day").steps);
+    }
+    if (boxes.batt) {
+      boxes.batt.lastLevel = E.getBattery();
+      boxes.batt.string = formatStr(boxes.batt, boxes.batt.lastLevel);
+      boxes.batt.lastUpdate = Date.now();
+    }
+
     Bangle.setUI({
-      mode : "clock",
-      remove : function() {
-        // Remove event handlers, stop draw timer, remove custom font if used
+      mode: "clock",
+      remove: function() {
+        // Remove event handlers, stop draw timer, remove custom font
         Bangle.removeListener('touch', touchHandler);
         Bangle.removeListener('drag', dragHandler);
+        Bangle.removeListener('step');
+        Bangle.removeAllListeners('lock');
```

Comment:
This seems a little agressive to me. Could you refactor to just remove this apps listener?

Reply:
I’ve organized all the event handler functions into section 3 and revised the cleanup in setUI. Thank you for catching the lock listener since that wasn’t supposed to be a remove all. And the remove step listener was missing the function to be removed. Do these changes look sufficient?